#! /bin/sh /usr/share/dpatch/dpatch-run
## 99_fix_build_with_gcc6.dpatch
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: commit a18b8e55c78f735446db69010872b66e8fbd5ff4
## DP: Author: Michael Vogt <mvo@ubuntu.com>
## DP: Date:   Thu Jan 21 16:59:53 2016 +0100
## DP: 
## DP:     Fix build with gcc6
## DP:     
## DP:     The make_pair() functions does not take a template parameter
## DP:     anymore.
## DP:     
## DP:     Thanks: Martin Michlmayer
## DP:     Closes: #811857

@DPATCH@
diff --git a/gtk/rglogview.cc b/gtk/rglogview.cc
index b55bc8da..317457d2 100644
--- a/gtk/rglogview.cc
+++ b/gtk/rglogview.cc
@@ -23,6 +23,7 @@
 #include <cassert>
 #include <cstring>
 #include <map>
+#include <utility>
 #include <apt-pkg/fileutl.h>
 
 #include "config.h"
@@ -86,7 +87,7 @@ void RGLogView::readLogs()
 			     COLUMN_LOG_TYPE, LOG_TYPE_TOPLEVEL, 
 			     -1);
 	 g_free(sort_key);
-	 history_map.insert(make_pair<int,GtkTreeIter>(history_key,month_iter));
+	 history_map.insert(std::make_pair(history_key,month_iter));
       } else {
 	 month_iter = history_map[history_key];
       }
